source/js/read-recipe.js

```diff
@@ -2,99 +1,99 @@
+  // <div class="each-ingredient">
+  //   <label class = "container">duck 0.5 lb
+  //     <input type="checkbox">
+  //     <span class="checkmark"></span>
+  //   </label>
+  // </div>
+  const ingList = document.getElementById("ingredient-list");
+  for (const ingredient of recipe["ingredients"]) {
+    const eachIng = document.createElement("div");
+    eachIng.classList.add("each-ingredient");
+    const label = document.createElement("label");
+    label.innerText = `${ingredient["name"]} ${ingredient["amount"]}`;
+    label.classList.add("container");
+    const input = document.createElement("input");
+    input.type = "checkbox";
+    const span = document.createElement("span");
+    span.classList.add("checkmark");
+    label.append(input, span);
+    eachIng.appendChild(label);
+    ingList.appendChild(eachIng);
   }
-
-  const recipeSteps = recipe["steps"];
-  const stepElems = document
-    .getElementById("instructions")
-    .getElementsByClassName("step-instruction");
-  for (let i = 0; i < stepElems.length; i++) {
-    const stepElem = stepElems[i];
-    const recipeStep = recipeSteps[i];
-    stepElem.querySelector(".card-subtitle").innerText = `Step ${i + 1}`;
-    stepElem.querySelector(".card-text").innerText = recipeStep;
+  // add step list
+  const stepList = document.getElementById("steps-list");
+  const ul = document.createElement("ol");
+  for (const step of recipe["steps"]) {
+    ul.classList.add("orderList");
+    const li = document.createElement("li");
+    li.innerHTML = step;
+    ul.appendChild(li);
   }
-
-  /*
-  const ingredients = [];
-  recipe.ingredients = ingredients;
-  const ingredient1 = {};
-  ingredient1.name = document.getElementById('ingredient1name').value.trim();
-  ingredient1.amount = document.getElementById('ingredient1amount')
-      .value.trim();
-  ingredients.push(ingredient1);
-  const steps = [];
-  recipe.steps = steps;
-  document.getElementById('step1')
-  document.getElementById('step2')
-  */
+  stepList.appendChild(ul);
 }
 
-/** Checks if ID is in localStorage */
-function checkID() {
-  const queryString = window.location.search;
-  // console.log(queryString);
-  const urlParams = new URLSearchParams(queryString);
-  const id = urlParams.get("id");
-  if (id === null) {
-    console.log("No id parameter");
-    return;
-  }
-  console.log(`id: ${id}`);
-  populateHTML(id);
+/** Sets event listeners */
+function setButtonListener() {
+  const editButton = document.getElementById("Edit");
+  editButton.addEventListener("click", (e) => {
+    location.href = `update-recipe.html?id=${recipeID}`;
+  });
+
+  const deleteButton = document.getElementById("Delete");
+  deleteButton.addEventListener("click", (e) => {
+    delete recipes[recipeID];
+    localStorage.setItem("recipes", JSON.stringify(recipes));
+    window.alert("successfully deleted the recipe!");
+    let currUrl = location.toString().replace("read-recipe.html?id=" + recipeID, "cook-book.html");
```

Comment:
Codacy has a fix for the issue: ['currUrl' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/cse110-fa21-group7/cse110-fa21-group7/pullRequest?prid=8446960)
```suggestion
    const currUrl = location.toString().replace("read-recipe.html?id=" + recipeID, "cook-book.html");
```